packages/insured-bridge-relayer/src/Relayer.ts

```diff
@@ -56,73 +64,165 @@ export class Relayer {
+      deposit.amount,
+      deposit.slowRelayFeePct,
+      deposit.instantRelayFeePct,
+      deposit.quoteTimestamp,
+      realizedLpFee
+    );
+
+    await this.gasEstimator.update();
+    try {
+      const { receipt, transactionConfig } = await runTransaction({
+        web3: this.l1Client.l1Web3,
+        transaction: (slowRelayTx as unknown) as TransactionType,
+        transactionConfig: { gasPrice: this.gasEstimator.getCurrentFastPrice().toString(), from: this.account },
+        availableAccounts: 1,
+      });
+
+      if (receipt.events)
+        this.logger.info({
+          at: "InsuredBridgeRelayer#Relayer",
+          type: "Slow Relay executed  🐌",
+          tx: receipt.transactionHash,
+          depositId: receipt.events.DepositRelayed.returnValues.depositId,
+          sender: receipt.events.DepositRelayed.returnValues.sender,
+          slowRelayer: receipt.events.DepositRelayed.returnValues.slowRelayer,
+          depositTimestamp: receipt.events.DepositRelayed.returnValues.depositTimestamp,
+          recipient: receipt.events.DepositRelayed.returnValues.recipient,
+          l1Token: receipt.events.DepositRelayed.returnValues.l1Token,
+          amount: receipt.events.DepositRelayed.returnValues.amount,
+          slowRelayFeePct: receipt.events.DepositRelayed.returnValues.slowRelayFeePct,
+          instantRelayFeePct: receipt.events.DepositRelayed.returnValues.instantRelayFeePct,
+          quoteTimestamp: receipt.events.DepositRelayed.returnValues.quoteTimestamp,
+          realizedLpFeePct: receipt.events.DepositRelayed.returnValues.realizedLpFeePct,
+          priceRequestAncillaryDataHash: receipt.events.DepositRelayed.returnValues.priceRequestAncillaryDataHash,
+          depositHash: receipt.events.DepositRelayed.returnValues.depositHash,
+          depositContract: receipt.events.DepositRelayed.returnValues.depositContract,
+          transactionConfig,
+        });
+      else throw receipt;
+    } catch (error) {
+      this.logger.error({ at: "InsuredBridgeRelayer#Relayer", type: "Something went wrong slow relaying!", error });
+    }
   }
 
-  private async fastRelay(deposit: Deposit) {
-    console.log(deposit);
+  private async speedUpRelay(/* deposit: Deposit*/) {
+    // TODO: implement
   }
 
-  private async fetchBotL1TokenBalance(l1TokenAddress: string): Promise<string> {
-    const l1Token = (new this.web3.eth.Contract(
-      getAbi("ExpandedERC20"),
-      l1TokenAddress
-    ) as unknown) as ExpandedERC20Web3;
+  private async instantRelay(/* deposit: Deposit*/) {
+    // TODO: implement
+  }
 
-    return await l1Token.methods.balanceOf(this.account).call();
+  private getRelayTokenRequirement(
+    deposit: Deposit,
+    proposerBondPct: BN,
+    realizedLpFeePct: BN
+  ): { slow: BN; instant: BN } {
+    // bridged amount - the LP fee, - slow relay fee, - instant relay fee
+    return {
+      slow: toBN(deposit.amount).mul(proposerBondPct).muln(2).div(fixedPointAdjustment),
```

Comment:
Why divide proposer bond by 2?

Reply:
ye, that's a mistake. I've fixed it an added additional comments.